docs/architecture/psa-migration/testing.md

```diff
@@ -0,0 +1,95 @@
+Testing strategy for `MBEDTLS_USE_PSA_CRYPTO`
+=============================================
+
+This document records the testing strategy used so far in implementing
+`MBEDTLS_USE_PSA_CRYPTO`.
+
+
+General considerations
+----------------------
+
+There needs to be at least one build in `all.sh` that enables
+`MBEDTLS_USE_PSA_CRYPTO` and runs the full battery of tests. Currently that's
+ensured by the fact that `scripts/config.py full` enables
+`MBEDTLS_USE_PSA_CRYPTO`.
+
```

Comment:
And there also needs to have builds with `MBEDTLS_USE_PSA_CRYPTO` _disabled_. (It's debatable whether `MBEDTLS_USE_PSA_CRYPTO` should be in `full` or not since it's a behavior change, not an extra option. IIRC we've discussed that in the past and decided to include it in `full` because it made some variety, with many builds based on `full` and so having it enabled _and_ many builds based on the default and so having it disabled.)